code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -253,8 +254,11 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 		if(blocked)
 			to_chat(H, "<span class='warning'>\The [blocked] is in the way!</span>")
 			return 1
-		H.visible_message("<span class='notice'>[H.name] takes a drag of their [name].</span>")
-		playsound(H, 'sound/items/cigs_lighters/inhale.ogg', 50, 0, -1)
+		if(last_drag <= world.time - 30) //Spam limiter. Only for messages/sound. 
+			last_drag = world.time
+			H.visible_message("<span class='notice'>[H.name] takes a drag of their [name].</span>")
+			playsound(H, 'sound/items/cigs_lighters/inhale.ogg', 50, 0, -1)
+		reagents.trans_to_mob(H, pick(1,1.5,2), CHEM_BREATHE) //Smokes it faster. Slightly random amount. 
```

Comment:
instead of flat numbers, is it possible to have a range?

Reply:
reagents.trans_to_mob(H, (rand(10,20)/10), CHEM_BREATHE) Should work, I think

Reply:
yeah sure aight